### PR TITLE
feat(schema-utils): Allow any input for json parsing

### PR DESCRIPTION
**Pull request by zharinov**
## Changes

For convenience purposes, `parseJson` and `safeParseJson` should allow for any input.
When the input is string, `JSON.parse` is applied.

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/util/schema-utils.ts [hunk 1]
```diff
@@ -94,16 +94,19 @@ export function looseValue<T, U extends z.ZodTypeDef, V>(
 export function parseJson<
   T = unknown,
   Schema extends z.ZodType<T> = z.ZodType<T>
->(input: string, schema: Schema): z.infer<Schema> {
-  const parsed = JSON.parse(input);
+>(input: unknown, schema: Schema): z.infer<Schema> {
+  let parsed = input;
+  if (typeof input === 'string') {
```
Comment (viceice) [inline lib/util/schema-utils.ts:99]: shouldn't we throw if it's not a string?

**Review by viceice — COMMENTED** [review]
otherwise LGTM

File: lib/util/schema-utils.ts (hunk 1 above)
Comment (zharinov) [reply lib/util/schema-utils.ts:99]: If provided non-string value doesn't match the schema, it will throw

Comment (viceice) [reply lib/util/schema-utils.ts:99]: then the function name is misleading. as we just validate the value against the schema when not string.

Comment (zharinov) [reply lib/util/schema-utils.ts:99]: Schema can transform data, in this case it can be called parsing

Comment (viceice) [reply lib/util/schema-utils.ts:99]: if you plan to use it for non json string input, then we should rename it. 

if we only use it to parse and validate json strings, then we should validate input to be string and throw if it's not string.

**Review by viceice — COMMENTED** [review]
see comment

**Comment by zharinov** [comment]
Actually, I'm not that interested in this PR

**Comment by zharinov** [comment]
Here is more elegant approach: https://github.com/renovatebot/renovate/pull/21536

**Pull request closed without merging**